### PR TITLE
Makes linter happy

### PR DESCRIPTION
**Pull request by aisk**
Since python does not have block scope but only function scope, current codes runs fine, but for static anylizers / linters, they think this may have issues. The modified codes have the same behaviors, but have more friendly for the linters.

### CODE REVIEW

**Review by wjblanke — DISMISSED** [review]
aok

**Comment by wjblanke** [comment]
> aok

Unfortunately this PR has made our linter unhappy :-(

Run mypy --exclude config.py tests
chia/server/address_manager.py:446: error: Need type annotation for 'cached_tried_matrix_positions' (hint: "cached_tried_matrix_positions: List[<type>] = ...")
chia/server/address_manager.py:476: error: Need type annotation for 'cached_new_matrix_positions' (hint: "cached_new_matrix_positions: List[<type>] = ...")
Found 2 errors in 1 file (checked 84 source files)
Error: Process completed with exit code 1.

Maybe a lint version issue?

**Comment by aisk** [comment]
@wjblanke The linters are happ now again😁

File: chia/server/address_manager.py [hunk 1]
```diff
@@ -443,8 +443,9 @@ def select_peer_(self, new_only: bool) -> Optional[ExtendedPeerInfo]:
         if not new_only and self.tried_count > 0 and (self.new_count == 0 or randrange(2) == 0):
             chance = 1.0
             start = time.time()
+            cached_tried_matrix_positions: List[Tuple[int, int]] = []
             if len(self.used_tried_matrix_positions) < math.sqrt(TRIED_BUCKET_COUNT * BUCKET_SIZE):
-                cached_tried_matrix_positions = list(self.used_tried_matrix_positions)
+                cached_tried_matrix_positions += self.used_tried_matrix_positions
```
Comment (arvidn) [inline chia/server/address_manager.py:448]: as someone who isn't a python expert, this is not obviously correct. If `used_tried_matrix_position` is iterable, the `list()` constructor will convert it to a list, whereas `+=` will just append it. Actually, the exact semantics of the `+=` operator when the right hand side is an iterable is not trivial to find documentation for.

As far as I can tell, the `+=` operator on lists behaves different than `A = A + B`, which seems quite unintuitive.

I would lean towards leaving these assignment as they were, but still declare the empty list above.

Comment (aisk) [reply chia/server/address_manager.py]: hi @arvidn, updated.

**Review by arvidn — APPROVED** [review]

**Pull request merged**